task_sdk/src/airflow/sdk/execution_time/context.py

```diff
@@ -0,0 +1,70 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+
+import structlog
+
+if TYPE_CHECKING:
+    from airflow.sdk.execution_time.comms import ConnectionResult
+
+
+def _convert_connection_result_conn(conn_result: ConnectionResult):
+    from airflow.sdk.definitions.connection import Connection
+
+    return Connection(**conn_result.model_dump(exclude={"type"}, by_alias=True))
+
+
+def get_connection(conn_id: str):
```

Comment:
```suggestion
def get_connection(conn_id: str) -> Connection | None:
```

Which also makes me wonder how we handle conn not found?

Reply:
Is this meant to be called but end users? Prefix it with _ otherwise?

Reply:
>Is this meant to be called but end users? Prefix it with _ otherwise?

Added `_` prefix.

>Which also makes me wonder how we handle conn not found?

I am planning to handle it [here](https://github.com/apache/airflow/blob/304140b4d08c6f88e0480a88cc0ed0f32469941c/task_sdk/src/airflow/sdk/execution_time/context.py#L68-L71) :

https://github.com/apache/airflow/blob/304140b4d08c6f88e0480a88cc0ed0f32469941c/task_sdk/src/airflow/sdk/execution_time/context.py#L68-L71

Reply:
@ashb Handled in https://github.com/apache/airflow/pull/45043/commits/38c65655f95a29288976049cd5037be6aed3a0d6